From 7828ff251d52867cd4ce4d413bc6fc28c0bc8120 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Fri, 21 Aug 2015 14:19:27 +0800
Subject: [PATCH 0268/5242] MLK-11395-3 regulator: consumer: add new event
 macros

commit  bbdf04cac4798da88180652d3f07502cb7d8f470 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add two new regulator events macro 'REGULATOR_EVENT_PRE_DO_ENABLE'
and 'REGULATOR_EVENT_PRE_DO_DISABLE', since some gpc operations
should be required when MIPI PHY is powered on/off.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/core.c           |    2 ++
 include/linux/regulator/consumer.h |    2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index cc1450c..83a8c9b 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -2143,6 +2143,7 @@ static int _regulator_do_enable(struct regulator_dev *rdev)
 {
 	int ret, delay;
 
+	_notifier_call_chain(rdev, REGULATOR_EVENT_PRE_DO_ENABLE, NULL);
 	/* Query before enabling in case configuration dependent.  */
 	ret = _regulator_get_enable_time(rdev);
 	if (ret >= 0) {
@@ -2283,6 +2284,7 @@ static int _regulator_do_disable(struct regulator_dev *rdev)
 {
 	int ret;
 
+	_notifier_call_chain(rdev, REGULATOR_EVENT_PRE_DO_DISABLE, NULL);
 	trace_regulator_disable(rdev_get_name(rdev));
 
 	if (rdev->ena_pin) {
diff --git a/include/linux/regulator/consumer.h b/include/linux/regulator/consumer.h
index 25602af..8fe6df70 100644
--- a/include/linux/regulator/consumer.h
+++ b/include/linux/regulator/consumer.h
@@ -121,6 +121,8 @@
 #define REGULATOR_EVENT_PRE_DISABLE		0x400
 #define REGULATOR_EVENT_ABORT_DISABLE		0x800
 #define REGULATOR_EVENT_ENABLE			0x1000
+#define REGULATOR_EVENT_PRE_DO_ENABLE		0x2000
+#define REGULATOR_EVENT_PRE_DO_DISABLE		0x4000
 
 /*
  * Regulator errors that can be queried using regulator_get_error_flags
-- 
1.7.9.5

